pkg/commands/status_discovery.go

```diff
@@ -53,19 +61,51 @@ func (s *CloudDiscoveryStatusStore) GetApiIngressStatus(cluster *kops.Cluster) (
 
 	if awsCloud, ok := cloud.(awsup.AWSCloud); ok {
 		name := "api." + cluster.Name
-		lb, err := awstasks.FindLoadBalancerByNameTag(awsCloud, name)
-		if lb == nil {
-			return nil, nil
+
+		ELB, NLB := false, false
+
+		lbSpec := cluster.Spec.API.LoadBalancer
+		switch lbSpec.Class {
+		case kops.LoadBalancerClassClassic, "":
+			ELB = true
+		case kops.LoadBalancerClassNetwork:
+			NLB = true
+		default:
+			return nil, fmt.Errorf("Unknown Cluster.Spec.API.LoadBalancer.Class : %v", lbSpec.Class)
 		}
-		if err != nil {
-			return nil, fmt.Errorf("error looking for AWS ELB: %v", err)
+
+		var lb interface{}
+
+		if ELB {
+			lb, err = awstasks.FindLoadBalancerByNameTag(awsCloud, name)
+			if isNil(lb) {
+				return nil, nil
+			}
+			if err != nil {
+				return nil, fmt.Errorf("error looking for AWS ELB: %v", err)
+			}
+		} else if NLB {
+			lb, err = awstasks.FindNetworkLoadBalancerByNameTag(awsCloud, name)
+			if isNil(lb) {
+				return nil, nil
+			}
+			if err != nil {
+				return nil, fmt.Errorf("error looking for AWS ELB: %v", err)
+			}
+
 		}
 		var ingresses []kops.ApiIngressStatus
 
-		if lb != nil {
-			lbDnsName := aws.StringValue(lb.DNSName)
+		if !isNil(lb) {
+			var lbDnsName string
+			if ELB {
+				lbDnsName = aws.StringValue(lb.(*elb.LoadBalancerDescription).DNSName)
```

Comment:
Probably better to pull the scope of `lb` `StringValue` call in the previous if/else blocks, so lb can be of the appropriate type. Pull `lbDnsName` to before the previous if/else block.